### PR TITLE
Add with options

### PR DESCRIPTION
**Pull request by groony**
Add ability to set and override options for `#with` to schema.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,6 +230,8 @@ module Users
   end
 end
 
+You can set options for `#with` to schema by overriding `#with_options`
```
Comment (kimrgrey) [inline README.md]: Please, provide here some clean and useful scenario from real life usage.

File: lib/performify/base.rb [hunk 2]
```diff
@@ -70,7 +70,9 @@ def fail?
       @result.blank?
     end
 
-    private def prepare_instance
+    private
```
Comment (kimrgrey) [inline lib/performify/base.rb]: Please, do not make code style changes: `private` is supposed to be attached to the method here, it was not a mistake.

File: lib/performify/validation.rb [hunk 3]
```diff
@@ -49,6 +49,12 @@ def errors
       def errors?
         errors.any?
       end
+
+      private
```
Comment (kimrgrey) [inline lib/performify/validation.rb]: Please, attach `private` to method.

File: README.md [hunk 4]
```diff
@@ -242,6 +242,32 @@ else
 end
 ```
 
+You can set options for `#with` to schema by overriding `#with_options`.
```
Comment (kimrgrey) [inline README.md]: Not sure if I can override `private` method in subclasses. Can I? Please, provide specs for both default and overridden case.

Comment (groony) [reply README.md]:
```
irb(main):039:0> class A
irb(main):040:1>   private def test
irb(main):041:2>     'hello'
irb(main):042:2>   end
irb(main):043:1> end
=> A
irb(main):044:0>
irb(main):045:0* class B < A
irb(main):046:1>
irb(main):047:1*   def show
irb(main):048:2>   test
irb(main):049:2>   end
irb(main):050:1>
irb(main):051:1*   private def test
irb(main):052:2>     puts "#{super} man"
irb(main):053:2>   end
irb(main):054:1> end
=> B
irb(main):055:0> B.new.show
hello man
=> nil
```

**Comment by groony** [comment]
any comments?

**Comment by kimrgrey** [comment]
Errors from Rubocop are going to be fixed in main repo.

**Comment by kimrgrey** [comment]
Fixed validation rules, merged and released as 0.8.0. Thanks for the contribution 👍

**Pull request merged**